clientd3d/d3drender.c

```diff
@@ -477,6 +479,195 @@ void D3DRenderShutDown(void)
 	}
 }
 
+
+// Global light scale (apply a percentage to all DLIGHT_SCALE results).
+// Change at runtime with SetGlobalLightScale().
+static float gLightScale = 0.45f;
+
+// Minimum world light radius in FINENESS units (1024 = one grid square).
+// With the bounding box check in D3DRenderLMapPostFloorAdd/Ceiling/Wall, we no longer
+// need a large minimum radius to catch lights centered in polygons. The bounding box
+// check handles that case. This minimum is just a safety floor for very small lights.
+// Set to 0 to allow full control via gLightScale and object intensity.
+static const float MIN_WORLD_LIGHT_RADIUS = 0.0f;
+
+void SetGlobalLightScale(float scale)
+{
+   // clamp to sensible range
+   if (scale < 0.0f)
+      scale = 0.0f;
+   if (scale > 5.0f)
+      scale = 5.0f;
+   gLightScale = scale;
+
+   // Force a full world rebuild so cached lightmaps / geometry that depend on radii update.
+   gD3DRedrawAll |= D3DRENDER_REDRAW_ALL;
+}
+
+float GetGlobalLightScale(void) { return gLightScale; }
+
+
+void D3DRenderDebugLightPositions(Draw3DParams *params)
+{
+   // Debug: Draw WIREFRAME ELLIPSOIDS oriented by sun direction at static light positions
+   const int LAT_SEGMENTS = 8;
+   const int LON_SEGMENTS = 12;
+
+   debug(("DEBUG: Drawing %d static lights as directional ellipsoids\n", gDLightCache.numLights));
+
+   // Get sun direction for orientation
+   const Vector3D &sunVect = getSunVector();
+   float sunAngle = atan2f(sunVect.y, sunVect.x);  // Angle in XY plane
+
+   debug(("  Sun vector: (%.2f, %.2f, %.2f), angle: %.2f rad\n", sunVect.x, sunVect.y, sunVect.z, sunAngle));

```

Comment:
Do these debug prints happen every frame?  Seems like a lot.  Are you going to remove them before checking in?